src/Compilers/CSharp/Test/Semantic/Semantics/Utf8StringsLiteralsTests.cs

```diff
@@ -0,0 +1,1790 @@
+            CompileAndVerify(comp, expectedOutput: @"
+() => Convert(new [] {104, 101, 108, 108, 111}, C1)
+() => Convert(new Span`1(new [] {100, 111, 103}), C2)
+() => Convert(new ReadOnlySpan`1(new [] {99, 97, 116}), C3)
+").VerifyDiagnostics();
+        }
+
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void OverloadResolution_01()
+        {
+            var source = @"
+using System;
+class C
+{
+    static void Main()
+    {
+        System.Console.WriteLine(Test(""s""));
+    }
+
+    static string Test(ReadOnlySpan<char> a) => ""ReadOnlySpan"";
+    static string Test(byte[] a) => ""array"";
+}
+";
+            var comp = CreateCompilation(source, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.DebugExe);
+
+            CompileAndVerify(comp, expectedOutput: @"ReadOnlySpan").VerifyDiagnostics();
+        }
+
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void OverloadResolution_02()
+        {
+            var source = @"
+using System;
+class C
+{
+    static void Main()
+    {
+        System.Console.WriteLine(Test(""s""));
+    }
+
+    static string Test(byte[] a) => ""array"";
+    static string Test(ReadOnlySpan<char> a) => ""ReadOnlySpan"";
+}
+";
+            var comp = CreateCompilation(source, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.DebugExe);
+
+            CompileAndVerify(comp, expectedOutput: @"ReadOnlySpan").VerifyDiagnostics();
+        }
+
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void OverloadResolution_03()
+        {
+            var source = @"
+class C
+{
+    static void Main()
+    {
+        System.Console.WriteLine(Test(""s""));
+    }
+
+    static string Test(C1 a) => ""string"";
+    static string Test(C2 a) => ""array"";
+}
+
+class C1
+{
+    public static implicit operator C1(string x)
+    {
+        return new C1();
+    }
+}
+
+class C2
+{
+    public static implicit operator C2(byte[] x)
+    {
+        return new C2();
+    }
+}
+";
+            var comp = CreateCompilation(source, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.DebugExe);
+
+            CompileAndVerify(comp, expectedOutput: @"string").VerifyDiagnostics();
+        }
+
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void OverloadResolution_04()
+        {
+            var source = @"
+using System;
+class C
+{
+    static void Main()
+    {
+        System.Console.WriteLine(Test(""s"", (int)1));
+    }
+
+    static string Test(ReadOnlySpan<char> a, long x) => ""ReadOnlySpan"";
+    static string Test(byte[] a, int x) => ""array"";
+}
+";
+            var comp = CreateCompilation(source, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.DebugExe);
+
+            CompileAndVerify(comp, expectedOutput: @"ReadOnlySpan").VerifyDiagnostics();
+        }
```

Comment:
Should there also be a test where a type defines both an implicit conversion from `string` and from e.g. `ReadOnlySpan<byte>`? Apologies if I missed it.

Reply:
> Should there also be a test where a type defines both an implicit conversion from ```string``` and from e.g. ```ReadOnlySpan<byte>```?

Having an additional test doesn't hurt. However, ```UserDefinedImplicitConversions_01``` demonstrates that a user defined conversion like that cannot be used as an implicit conversion.